File: cloud_dairy/lib/models/farmer.dart
class Farmer {
  final String id;
  final String name;
  final String phone;
  final String address;
  final double balance;
  final double advance;

  Farmer({
    required this.id,
    required this.name,
    required this.phone,
    required this.address,
    this.balance = 0.0,
    this.advance = 0.0,
  });

  factory Farmer.fromJson(Map<String, dynamic> json) {
    return Farmer(
      id: json['_id'],
      name: json['name'],
      phone: json['phone'],
      address: json['address'] ?? '',
      balance: (json['balance'] ?? 0).toDouble(),
      advance: (json['advance'] ?? 0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phone': phone,
      'address': address,
    };
  }
}

File: models/Farmer.js
const mongoose = require('mongoose');

const farmerSchema = new mongoose.Schema({
    name: { type: String, required: true },
    phone: { type: String, required: true, unique: true },
    address: { type: String },
    password: { type: String, required: true }, // For login
    balance: { type: Number, default: 0 },
    totalMilk: { type: Number, default: 0 },
    advance: { type: Number, default: 0 },
    lastPaymentDate: { type: Date },
    createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Farmer', farmerSchema);
